src/identity.cc

```diff
@@ -680,175 +694,288 @@ TRITONBACKEND_ModelInstanceExecute(
-      // memory but we have to handle any returned type. If we get
-      // back a buffer in GPU memory we just fail the request.
-      void* output_buffer;
-      TRITONSERVER_MemoryType output_memory_type = TRITONSERVER_MEMORY_CPU;
-      int64_t output_memory_type_id = 0;
+      const char* input_name;
+      TRITONSERVER_DataType input_datatype;
+      const int64_t* input_shape;
+      uint32_t input_dims_count;
+      uint64_t input_byte_size;
+      uint32_t input_buffer_count;
       GUARDED_RESPOND_IF_ERROR(
           responses, r,
-          TRITONBACKEND_OutputBuffer(
-              output, &output_buffer, input_byte_size, &output_memory_type,
-              &output_memory_type_id));
-      if ((responses[r] == nullptr) ||
-          (output_memory_type == TRITONSERVER_MEMORY_GPU)) {
-        GUARDED_RESPOND_IF_ERROR(
-            responses, r,
-            TRITONSERVER_ErrorNew(
-                TRITONSERVER_ERROR_UNSUPPORTED,
-                "failed to create output buffer in CPU memory"));
+          TRITONBACKEND_InputProperties(
+              input, &input_name, &input_datatype, &input_shape,
+              &input_dims_count, &input_byte_size, &input_buffer_count));
+      if (responses[r] == nullptr) {
         LOG_MESSAGE(
             TRITONSERVER_LOG_ERROR,
             (std::string("request ") + std::to_string(r) +
-             ": failed to create output buffer in CPU memory, error response "
-             "sent")
+             ": failed to read input properties, error response sent")
                 .c_str());
         continue;
       }
 
-      // Step 3. Copy input -> output. We can only handle if the input
-      // buffers are on CPU so fail otherwise.
-      size_t output_buffer_offset = 0;
-      for (uint32_t b = 0; b < input_buffer_count; ++b) {
-        const void* input_buffer = nullptr;
-        uint64_t buffer_byte_size = 0;
-        TRITONSERVER_MemoryType input_memory_type = TRITONSERVER_MEMORY_CPU;
-        int64_t input_memory_type_id = 0;
+      LOG_MESSAGE(
+          TRITONSERVER_LOG_INFO,
+          (std::string("\tinput ") + input_name + ": datatype = " +
+           TRITONSERVER_DataTypeString(input_datatype) + ", shape = " +
+           backend::ShapeToString(input_shape, input_dims_count) +
+           ", byte_size = " + std::to_string(input_byte_size) +
+           ", buffer_count = " + std::to_string(input_buffer_count))
+              .c_str());
+
+      // We validated that the model configuration specifies N inputs,
+      // but the request is not required to request any output at all
+      // so we only produce an output if requested.
+      std::string output_name = "OUTPUT" + std::string(input_name, 5, -1);
+      bool is_requested = false;
+      const char* requested_output_name = nullptr;
+      for (uint32_t io_index = 0; io_index < requested_output_count;
+           io_index++) {
+        if (output_name.compare(requested_output_names[io_index]) == 0) {
```

Comment:
if set.contains(output_name) then can just continue? No need for is_requested flag?

Reply:
I think you should handle this backward, instead of iterating "input -> check if corresponding output is requested -> copy to output", you should just iterate "requested output -> look up corresponding input -> copy to output".

Then you don't need to maintain a set of requested output

Reply:
Yes, thats a good idea. Did that